rfd/0153-resource-guidelines.md

```diff
@@ -480,64 +480,123 @@ func (s *FooService) ListFoos(ctx context.Context, pageSize int, pageToken strin
 
 ### Cache
 
-One thing to consider when creating a resource is whether it will need to be cached. As mentioned above, any resource
-that is cached must have its backend layer implement the specific set of operations required by the cache collections
-[executor](https://github.com/gravitational/teleport/blob/004d0db0c1f6e9b312d0b0e1330b6e5bf1ffef6e/lib/cache/collections.go#L54-L76).
-While `Upsert` and `DeleteAll` semantics are required by the cache it is preferred that the two methods are not directly
-exposed in the gRPC API. Several existing resources include a `DeleteAll` purely for the cache that always returns a
-`trace.NotImplemented` error. To avoid exposing the methods in the gRPC API at all, a local variant of the backend
-service similar
-to [`services.DynamicAccessExt`](https://github.com/gravitational/teleport/blob/004d0db0c1f6e9b312d0b0e1330b6e5bf1ffef6e/lib/services/access_request.go#L260-L278)
-should be used.
-
 Caching is most important for resources that are accessed frequently and in a "hot path" (i.e., during the process of
 performing normal day-to-day operations). For example, resources like cluster networking config, session recording
 config, CAs, roles, etc. which are retrieved per connection should be cached to reduce latency. Resources which are
 accessed infrequently, or which scale linearly with cluster size are good examples of resources that should NOT be
 cached.
 
-If a resource is to be cached, it must be added to
-the [Auth cache](https://github.com/gravitational/teleport/blob/004d0db0c1f6e9b312d0b0e1330b6e5bf1ffef6e/lib/cache/cache.go#L95-L154)
-and the cache of any service that requires it. To add the `Foo` resource to the cache its executor would look similar
-to the following:
+If a resource is to be cached, it must be added to the
```

Comment:
Do we want to mention what to do if a resource must be watchable but doesn't have to be cached?